Sources/Datadog/Core/Persistence/FilesOrchestrator.swift

```diff
@@ -11,11 +11,25 @@ internal struct WritableFileConditions {
     let maxFileSize: UInt64
     let maxFileAgeForWrite: TimeInterval
     let maxNumberOfUsesOfFile: Int
+    let maxWriteSize: UInt64
+
+    init(performance: PerformancePreset) {
+        self.maxDirectorySize = performance.maxSizeOfLogsDirectory
+        self.maxFileSize = performance.maxBatchSize
+        self.maxFileAgeForWrite = performance.maxFileAgeForWrite
+        self.maxNumberOfUsesOfFile = performance.maxLogsPerBatch
+        self.maxWriteSize = performance.maxLogSize
+    }
```

Comment:
Thanks @buranmert 💪. I understand this idea of decomposing, but for now I think we won't leverage much with this. Two main reasons:
* current set of environments (`.iOSApp` and `.iOSAppExtension`) doesn't require further decomposition of the `PerformancePreset`;
* a crucial thing to any `PerformancePreset` is its consistency, so that uploads can seamlessly follow writes and reads. To ensure this, `PerformancePresetTests` was created. If we decompose `PerformancePreset` into chunks, we open it for any possible permutation and the validation of preset consistency should happen at runtime. Now, as it's declared in one place, it is validated at compile time (and asserted in unit tests).